core/src/main/java/jenkins/views/Header.java

```diff
@@ -54,4 +62,28 @@ public static Header get() {
         return header.orElseGet(JenkinsHeader::new);
     }
 
+    /**
+     * @return a list of {@link Action} to show in the header, defaults to {@link hudson.model.RootAction} extensions
+     */
+    @Restricted(NoExternalUse.class)
+    public List<Action> getActions() {
+        // There's an issue where new actions (e.g. a new plugin installation) don't appear in the order
+        // of their ordinal annotation - to work around that we manually sort the list
+        Map<String, Double> rootActionsOrdinal = ExtensionList.lookup(RootAction.class)
+                .getComponents()
+                .stream()
+                .collect(Collectors.toMap(
+                        c -> c.getInstance().getClass().getName(),
+                        ExtensionComponent::ordinal
+                ));
+
+        return Jenkins.get()
+                .getActions()
+                .stream()
+                .filter(e -> e.getIconFileName() != null)
```

Comment:
This filtering is not correct. Plugins might return `null` and use `getIconClassName` instead when they implement `IconSpec`.  E.g. PrioritySorter plugin does it this way.
```suggestion
                .filter(e -> e.getIconFileName() != null || (e instanceof IconSpec is && is.getIconClassName() != null))
```